compiler/testData/diagnostics/tests/controlFlowAnalysis/enumInterdependence.fir.kt

```diff
@@ -1,10 +1,10 @@
 enum class A(val v: A) {
-    A1(A2),
+    A1(<!UNINITIALIZED_ENUM_ENTRY!>A2<!>),
     A2(A1)
 }
 
 enum class D(val x: Int) {
-    D1(D2.x),
+    D1(<!UNINITIALIZED_ENUM_ENTRY!>D2<!>.x),
```

Comment:
Here's another case which I think is missed in FE1.0 (causes NPE) but I believe you catch (`calleeEnumEntry == accessedContext`). Could you add and try:

```kt
enum class E(val x: Int) {
  D1(D1.x);
}
```

Reply:
Indeed! Added a self-reference case.